flytekit/engines/unit/engine.py

```diff
@@ -148,6 +148,16 @@ def _transform_for_user_output(self, outputs):
 
 class DynamicTask(ReturnOutputsTask):
 
+    def __init__(self, *args, **kwargs):
+        self._has_workflow_node = False
+        super(DynamicTask, self).__init__(*args, **kwargs)
+
+    def _transform_for_user_output(self, outputs):
+        if self.has_workflow_node:
+            # If a workflow node has been detected, then we skip any transformation
```

Comment:
unclear to me why...

Reply:
theoretically, we shouldn't need a check like this. Can you explain the reasoning a bit?

Reply:
I'm going to leave this here for now, if that's okay and add a lot of warning logs and comments.  I'm basically short-circuiting the unit test engine.  I want to be able to at least test the dynamic job spec that's produced, but don't have time to finish the rest of the unit test engine right now.  Is this okay?

Reply:
👍

Reply:
just add a `TODO:` so it searchable please

Reply:
oo you did below. nevermind.